### PR TITLE
adding tags attribute to KMS key resource

### PR DESCRIPTION
**Pull request by Rohit1509**
Signed-off-by: Rohit Joshi <rohit.prasad.joshi@sap.com>

### Description

Please describe what this change achieves. Ensure you have read the [Contributing to InSpec AWS](https://github.com/inspec/inspec-aws/CONTRIBUTING.md) document before submitting.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_kms_key.rb [hunk 1]
```diff
@@ -39,6 +39,24 @@ def exists?
...
+    end
+    tags
+  end
+
+  def tags
+    begin
+      tag_list = @aws.kms_client.list_resource_tags(key_id: @display_name).tags
+    rescue
+      return {}
+    end
```
[the remark this answers is not in the document]
Comment (SathishAchilles) [reply libraries/aws_kms_key.rb:56]: You are right about most existing resources using `begin; rescue` block, I am currently investigating that..

[the remark this answers is not in the document]
Comment (sathish-progress) [reply libraries/aws_kms_key.rb:56]: You are right about most existing resources using `begin; rescue` block, I am currently investigating that..

File: libraries/aws_kms_key.rb [hunk 2]
```diff
@@ -39,6 +39,24 @@ def exists?
...
+    kms_tags(tag_list)
+  end
```
Comment (sathish-progress) [inline libraries/aws_kms_key.rb:58]: actually, I tried the below, I was able to query the tags successfully, 
```suggestion
  def tags
    catch_aws_errors do
      tag_list = @aws.kms_client.list_resource_tags(key_id: @display_name).tags
      kms_tags(tag_list)
    end
  end
```

Comment (Rohit1509) [reply libraries/aws_kms_key.rb]: fixed

**Review by sathish-progress — APPROVED** [review]
LGTM

File: docs/resources/aws_kms_key.md [hunk 3]
```diff
@@ -56,6 +56,7 @@ See also the [AWS documentation on KS Keys](https://docs.aws.amazon.com/kms/late
 |description       | The description of the key. |
 |deletion\_time     | Specifies the date and time after which AWS KMS deletes the key. This value is present only when KeyState is PendingDeletion, otherwise this value is nil. |
 |invalidation\_time | Provides the date and time until the key is not valid.  Once the key is not valid, AWS KMS deletes the key and it becomes unusable.  This value will be null unless the keys Origin is EXTERNAL and its matcher have\_key\_expiration is set to true. |
+|tags          | An hash with each key-value pair corresponding to a tag associated with the entity |
```
Comment (IanMadd) [inline docs/resources/aws_kms_key.md:59]:
```suggestion
|tags          | A hash with each key-value pair corresponding to a tag associated with the entity. |
```

Comment (Rohit1509) [reply docs/resources/aws_kms_key.md]: fixed

**Comment by soumyo13** [comment]
@IanMadd Can you please approve if the review is done?

**Review by IanMadd — APPROVED** [review]

**Pull request merged**